ec2publishimg

```diff
@@ -28,226 +28,312 @@ from ec2imgutils.ec2imgutilsExceptions import (
+            secret_key=secret_key,
+            visibility=visibility,
+            log_callback=logger
+        )
+    except EC2PublishImgException as e:
         logger.error(e)
         sys.exit(1)
+    return publisher
 
-if not secret_key:
-    logger.error('Could not determine account secret access key')
-    sys.exit(1)
-
-regions = utils.get_regions(args, access_key, secret_key)
-snap_copy = args.allowCopy.lower()
-if (
-        snap_copy != 'image' and
-        snap_copy != 'none' and not
-        utils.validate_account_numbers(snap_copy)
-):
-    msg = 'Expecting "image", "none", or comma separated list of 12 digit AWS '
-    msg += 'account numbers as value of --allow-copy'
-    logger.error(msg)
-    sys.exit(1)
-
-visibility = args.share.lower()
-if (
-    visibility != 'all' and
-    visibility != 'none' and not
-    utils.validate_account_numbers(visibility)
-):
-    msg = 'Expecting "all", "none", or comma separated list of 12 digit AWS '
-    msg += 'account numbers as value of --share'
-    logger.error(msg)
-    sys.exit(1)
-
-if visibility[-1] == ',':
-    visibility = visibility[:-1]
-
-publisher = ec2pubimg.EC2PublishImage(
-        access_key=access_key,
-        allow_copy=snap_copy,
-        image_id=args.pubImgID,
-        image_name=args.pubImgName,
-        image_name_fragment=args.pubImgNameFrag,
-        image_name_match=args.pubImgNameMatch,
-        secret_key=secret_key,
-        visibility=visibility,
-        log_callback=logger
-)
 
-if args.dryRun:
-    logger.info('Dry run, image attributes will not be modified')
-try:
-    for region in regions:
-        publisher.set_region(region)
-        if args.dryRun:
+# ----------------------------------------------------------------------------
+def publish_images_in_region(publisher, dryRun, region, logger):
+    """Function to publish images in region"""
+    try:
+        if dryRun:
+            logger.info('Dry run, image attributes will not be modified')
             publisher.print_publish_info()
         else:
             publisher.publish_images()
-except EC2PublishImgException as e:
-    logger.exception(e)
-    sys.exit(1)
-except Exception as e:
-    logger.exception(e)
-    sys.exit(1)
+    except EC2PublishImgException as e:
+        logger.exception(e)
+        sys.exit(1)
+    except Exception as e:
+        logger.exception(e)
+        sys.exit(1)
+
+
+# ----------------------------------------------------------------------------
+def main(args):
+    args = parse_args(args)
+    logger = utils.get_logger(args.verbose)
+    check_required_arguments(args, logger)
+
+    config = get_config(args, logger)
+    access_key = get_access_key(args, config, logger)
+    secret_key = get_secret_key(args, config, logger)
+
+    regions = utils.get_regions(args, access_key, secret_key)
+    publisher = get_publisher(args, access_key, secret_key, logger)
+
+    for region in regions:
+        publisher.set_region(region)
```

Comment:
Seems like the `publish_images_in_region` function would handle setting the region on the class.

Reply:
Yep, you're right. It makes much more sense within the function. Done!